packages/entities/entities-plugins/src/components/free-form/shared/composables.ts

```diff
@@ -189,5 +213,248 @@ export function useFormShared<T>() {
+  ip: 'IP',
+  ssl: 'SSL',
+  ttl: 'TTL',
+  url: 'URL',
+  http: 'HTTP',
+}
+
+function replaceByDictionary(name: string) {
+  return labelDictionary[name.toLocaleLowerCase()] ?? name
+}
+
+export function useLabelPath(fieldName: string, rule: MaybeRefOrGetter<ResetLabelPathRule | undefined>) {
+  type Setting = {
+    parentPath: string | null
+    isolate: boolean
+  }
+
+  const inheritedSetting = inject<MaybeRefOrGetter<Setting>>(FIELD_RESET_LABEL_PATH_SETTING, {
+    parentPath: null,
+    isolate: false,
+  })
+
+  const { parentPath, isolate } = toValue(inheritedSetting)
+
+  // default inherit from parent
+  const finalPath = ref(parentPath ? pathUtils.resolve(parentPath, fieldName) : fieldName)
+
+  const nextSetting = ref<Setting>({
+    parentPath: finalPath.value,
+    isolate,
+  })
+
+  watch([toRef(rule), toRef(inheritedSetting)], ([ruleValue, settingValue]) => {
+    const { parentPath, isolate } = settingValue
+    const inheritedPath = parentPath ? pathUtils.resolve(parentPath, fieldName) : fieldName
+
+    if (ruleValue) {
+      const ruleHandlers: Record<ResetLabelPathRule, () => void> = {
+        'isolate': () => {
+          finalPath.value = fieldName
+          nextSetting.value.isolate = true
+          nextSetting.value.parentPath = null
+        },
+        'isolate-children': () => {
+          nextSetting.value.isolate = true
+          nextSetting.value.parentPath = null
+        },
+        'reset': () => {
+          finalPath.value = fieldName
+          nextSetting.value.isolate = false
+          nextSetting.value.parentPath = null
+        },
+        'reset-children': () => {
+          nextSetting.value.isolate = false
+          nextSetting.value.parentPath = null
+        },
+        'inherit': () => {
+          finalPath.value = inheritedPath
+          nextSetting.value.isolate = false
+          nextSetting.value.parentPath = inheritedPath
+        },
+      }
+
+      ruleHandlers[ruleValue]?.()
+    } else if (isolate) {
+      finalPath.value = fieldName
+      nextSetting.value.isolate = true
+      nextSetting.value.parentPath = null
+    }
+  }, { deep: true, immediate: true })
+
+  provide(FIELD_RESET_LABEL_PATH_SETTING, nextSetting)
+
+  return finalPath
+}
+
+export function useFieldLabel(
+  fieldPath: MaybeRefOrGetter<string>,
+  resetLabelPathRule: MaybeRefOrGetter<ResetLabelPathRule | undefined>,
+) {
+  const pathValue = toValue(fieldPath)
+  const fieldName = pathUtils.getName(pathValue)
+  const { formConfig } = useFormShared()
+  const parentLabelPath = useLabelPath(fieldName, resetLabelPathRule)
+
+  return computed(() => {
+    const realPath = parentLabelPath.value ?? fieldName
+    const parts = pathUtils.toArray(realPath)
+
+    const res = parts
+      .map(fieldName => fieldName
+        .replaceAll('_', ' ')
+        .split(' ')
+        .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+        .map(replaceByDictionary)
+        .join(' '))
```

Comment:
I am not sure where this is used, but seems like we can simplify this to 

```suggestion
        .split('_')
        .map(_.capitalize)
        .map(replaceByDictionary)
        .join(' '))
```